src/main/scala/loamstream/util/code/ScalaId.scala

```diff
@@ -150,7 +150,7 @@ sealed trait ChildId extends ScalaId {
   /** Id of the parent */
   def parent: ScalaId
 
-  def parentOpt: Some[ScalaId] = Some(parent)
+  override def parentOpt: Some[ScalaId] = Some(parent)
```

Comment:
From earlier in the review, I have no context for what `ScalaId`, `PackageId`, and `ObjectId` are. Can you describe them a little bit?

Reply:
They're representations of identifiers of Scala "entities" like objects, classes, and packages.  You can make the various Ids from each other, or from runtime- or compile-time-types, then use the Ids to generate code, like
```
package  foo.bar

class Baz

object Baz { }
```
later on,
```
val bazId = getScalaIdHoweverYouDo[foo.bar.Baz.type]
```
and then
```
val code: String = s"val soAndSo = ${bazId.inScalaFull}"

println(code) // prints "val soAndSo = foo.bar.Baz"
```
That's my high-level understanding.  Oliver wrote this code years ago; it's not central to this PR.